In [ ]:
!pip install tiktoken

In [5]:
import importlib
import tiktoken

print ('tiktoken version:', tiktoken.__version__)

tiktoken version: 0.11.0


--------------------------------------------

--------------------------------------------

instantiate the BPE tokenizer from tiktoken:

In [6]:
tokenizer = tiktoken.get_encoding('gpt2')

In [17]:
text = (
    "Hello, do you like tea? <|endoftext|> In the sunlit terraces of the palace"
    "of someunknownPlace"
)

integers = tokenizer.encode(text, allowed_special={"<|endoftext|>"})
print(integers)

[15496, 11, 466, 345, 588, 8887, 30, 220, 50256, 554, 262, 4252, 18250, 8812, 2114, 286, 262, 20562, 1659, 617, 34680, 27271]


In [18]:
strings = tokenizer.decode(integers)
print(strings

Hello, do you like tea? <|endoftext|> In the sunlit terraces of the palaceof someunknownPlace


-----------------------------------------------

Observations:
- first, the '<|endoftext|>' token is assigned a relatively large token ID, namely, 50256
- BPE tokenizer encodes and decodes unknown words, such as "someunknownPlace"

-----------------------------------------------